### PR TITLE
Feature: add the ability for a user to amend a commit

### LINKED ISSUE #15: Add the ability to amend a user submission
The submission process has most of the kinks worked out. The back half of the process that needs addressing is updating the PR based on triager feedback.

### LINKED ISSUE #29: Add schema version to submissions
Add `version: 2` to knowledge/skill submissions.

### PR DESCRIPTION
**Pull request by nerdalert**
Details in #15

Closes #15 
Closes #29

### CODE REVIEW

**Comment by nerdalert** [comment]
View the code running here for testing. Please try it if you have the time. http://34.229.17.203:4000/

Here is a populated dashboard:

image

File: .env.example [hunk 1]
```diff
@@ -12,3 +12,5 @@ GITHUB_TOKEN=<TOKEN FOR OAUTH INSTRUCTLAB MEMBER LOOKUP>
 TAXONOMY_REPO_OWNER=<GITHUB_ACCOUNT>
 TAXONOMY_REPO=<REPO_NAME>
```
Comment (toraponibm) [inline .env.example:13]: Are `TAXONOMY_REPO_OWNER` and `TAXONOMY_REPO` still in use after this PR?

Comment (nerdalert) [reply .env.example]: @toraponibm thanks for catching that. I had to switch the ENVs for be prefixed with `NEXT_PUBLIC_` for them to be accessible by the client side in NextJs.

**Review by toraponibm — APPROVED** [review]
LGTM

**Comment by juancappi** [comment]
I tested it to work on the CI environment. I do see a couple usability issues. Not sure if I should report them or it's assumed that it's a WIP and things are not polished yet. 

Given the scope is purely about amend, this LGTM.

**Comment by nerdalert** [comment]
> I tested it to work on the CI environment. I do see a couple usability issues. Not sure if I should report them or it's assumed that it's a WIP and things are not polished yet.
> 
> Given the scope is purely about amend, this LGTM.

@juancappi  looking to merge it. Throw them at me. We can do them on follow up smaller PRs. Not a huge fan of big PRs like this but since it's a feature it sprawled on me. Happy to open the issues, ty for spinning it up!

**Review by vishnoianil — APPROVED** [review]
LGTM

**Pull request merged**